### DISCUSSION TITLE
Expose generate Magic link function

### DISCUSSION THREAD

**Discussion by awaseem** [pbteja1998/remix-auth-email-link#15] [Ideas]
@pbteja1998 Any thoughts on exposing the magic link function publicly within the package? 

My use case is that I send users notifications when something changes in their account. I just want them to click the email link and be logged in right away. 

More than happy to create a PR, shouldn't be a complicated change.

**Comment by pbteja1998** [comment]
Hey @awaseem 

I am not completely sure what you mean. Why can't you use `authenticator.authenticate`? That would trigger the email, right?

But yeah, if you can open a PR, maybe that will make it more clear about what you have in mind.

**Comment by awaseem** [comment]
ahhh my bad @pbteja1998 you are right, calling `authenticator.authenticate` will send the email and I can add logic around the send email function as needed. Sorry ignore my late night ramblings

**Reply by pbteja1998** [reply]
Haha, no problem!

**Comment by lucascurti** [comment]
Actually, I think I have a use case for exposing `getMagicLink()` function. I'm creating Cypress tests for the login and I'm having a hard time finding the best way to do it. I don't really care about the actual email delivery (in fact the delivery is disabled in Cypress environment) but I would like to have somehow a way to grab the magic link. Still not ideal but, if the function is exposed, will make it possible to create a Cypress task to create the magic link for the environment.

What I'm doing right now is creating a magic link with a really long `linkExpirationTime` and `validateSessionMagicLink` set to `false` and seems to work but, since the magic link depends on the domain, I need to generate 1 magic link for each user for each environment (and I have users with different roles, so... more manually generated magic links 😅). It would be nice to be able to do this programmatically.

**Reply by pbteja1998** [reply]
Please feel free to open a PR.

**Reply by lucascurti** [reply]
Will do 😉

**Reply by lucascurti** [reply]
Here is the PR: https://github.com/pbteja1998/remix-auth-email-link/pull/16

**Reply by pbteja1998** [reply]
Published it at v1.4.2
